tests/cpu-vm

```diff
@@ -66,6 +72,21 @@ done
 lxc config set v1 limits.cpu="${cpuCount}"
 [ "$(lxc exec v1 -- ls /sys/devices/system/cpu | grep -Ec 'cpu[[:digit:]]+')" -eq "${cpuCount}" ]
 
+# check that CPU affinity is automatically set if feature present
+if journalctl --quiet --no-hostname --no-pager --boot=0 --unit=snap.lxd.daemon.service | grep "Scheduler: virtual-machine"; then
+  QEMU_PID=$(lxc info v1 | awk '/^PID:/ {print $2}')
+
+  # Check that there are processes with pinning set
+  # It will be shown like this:
+  # pid 2894's current affinity list: 6
+  # pid 2895's current affinity list: 8
+  # pid 2897's current affinity list: 0-15
+  # pid 2898's current affinity list: 0-15
+  # pid 2899's current affinity list: 0-15
+  # 2894 and 2895 have affinity set, while others don't
+  taskset --cpu-list -a -p "${QEMU_PID}" | grep -E ':\s+[0-9]+$'
+fi
```

Comment:
Am I right that there is a pinned tasks/threads for each vCPU given to the VM? If so, I'd go even further and count them and make sure it matches `limits.cpu`.

Also, since we are relying on a debug string to check for pining, I'd probably add an `else` branch that makes sure there is no pinning going on.

Reply:
Agree completely. Have fixed!